skypy/halos/tests/test_sham.py

```diff
@@ -0,0 +1,1209 @@
+
+        # Catalogue
+        parent_halo = 10**(np.array([10, 12, 11]))
+        z_halo = [0.05, 0.1, 0.01]
+        gen_sub_cat(parent_halo, z_halo, alpha, beta, gamma, x)
+        assert str(excinfo.value) == 'Subhalo alpha must be less than 2'
+
+    with pytest.raises(Exception) as excinfo:
+        # Parameters
+        alpha, beta, gamma, x = 1.91, 0.39, 0.1, 0.5
+
+        # Catalogue
+        parent_halo = 10**(np.array([10, 12, 11]))
+        z_halo = [0.05, 0.1, 0.01]
+        gen_sub_cat(parent_halo, z_halo, alpha, beta, gamma, x)
+        assert str(excinfo.value) == 'Subhalo x cannot be less than 1'
+
+    with pytest.raises(Exception) as excinfo:
+        # Parameters
+        alpha, beta, gamma, x = 1.91, -0.39, 0.1, 3
+
+        # Catalogue
+        parent_halo = 10**(np.array([10, 12, 11]))
+        z_halo = [0.05, 0.1, 0.01]
+        gen_sub_cat(parent_halo, z_halo, alpha, beta, gamma, x)
+        assert str(excinfo.value) == 'Subhalo beta must be between 0 and 1'
+
+    with pytest.raises(Exception) as excinfo:
+        # Parameters
+        alpha, beta, gamma, x = 1.91, 1.39, -0.1, 3
+
+        # Catalogue
+        parent_halo = 10**(np.array([10, 12, 11]))
+        z_halo = [0.05, 0.1, 0.01]
+        gen_sub_cat(parent_halo, z_halo, alpha, beta, gamma, x)
+        assert str(excinfo.value) == 'Subhalo beta must be between 0 and 1'
+
+    with pytest.raises(Exception) as excinfo:
+        # Parameters
+        alpha, beta, gamma, x = 1.91, 0.39, -0.1, 3
+
+        # Catalogue
+        parent_halo = 10**(np.array([10, 12, 11]))
+        z_halo = [0.05, 0.1, 0.01]
+        gen_sub_cat(parent_halo, z_halo, alpha, beta, gamma, x)
+        assert str(excinfo.value) == 'Subhalo gamma must be between 0 and 1'
+
+    with pytest.raises(Exception) as excinfo:
+        # Parameters
+        alpha, beta, gamma, x = 1.91, 1.39, 1.1, 3
+
+        # Catalogue
+        parent_halo = 10**(np.array([10, 12, 11]))
+        z_halo = [0.05, 0.1, 0.01]
+        gen_sub_cat(parent_halo, z_halo, alpha, beta, gamma, x)
+        assert str(excinfo.value) == 'Subhalo gamma must be between 0 and 1'
+
+
+@pytest.mark.flaky
+def test_galaxy_cat():
+    from astropy.cosmology import FlatLambdaCDM
+    import os
+    from skypy.halos.sham import galaxy_cat
+
+    # Parameters
+    m_star, phi_star, alpha = 10**(10.58), 10**(-2.77), -0.33
+    cosmology = FlatLambdaCDM(H0=70, Om0=0.3, name='FlatLambdaCDM')
+    z_range = [0.01, 0.1]
+    skyarea = 600
+    min_mass = 10**(7.5)
+    max_mass = 10**(14)
+    file_name = 'unit_test.yaml'
+
+    # Function
+    cat = galaxy_cat(m_star, phi_star, alpha, cosmology, z_range,
+                     skyarea, min_mass, max_mass, file_name)
+
+    # Check the catalogue
+    assert cat.size
+
+    # Check file exists
+    assert os.path.exists('unit_test.yaml')
+
+    # Open file and check structure
+    # Create expected lines
+    line1 = 'm_star: !numpy.power [10, 10.58]\n'

```

Comment:
See above comments on using the skypy pipeline machinery rather than internal yaml manipulation.